### PR TITLE
Fixed Extra Padding

### LINKED ISSUE #1820: Allow for specifying custom padding character
Currently, when using the `eng` formatter and specifying width 1 greater than actual number width, the number gets [padded with a space][1], which looks weird for some numbers:

```toml
[[block]]
block = "cpu"
format = " c$utilization.eng(width:2) "

[[block]]
block = "memory"
format = " m$mem_used_percents.eng(width:2) "
```

image

[1]: https://github.com/greshake/i3status-rust/blob/0f1322b165d91092366df11a5d3a9972f8f26d58/src/formatting/formatter.rs#L368

### PR DESCRIPTION
**Pull request by virajbhartiya**
Fixed #1820 

Fixed the extra padding issue

### CODE REVIEW

**Review by MaxVerevkin — CHANGES_REQUESTED** [review]
The padding is intentional. When `.eng(w:3)` is used, the space is inserted to prevent the block from changing it's size. The entire point of `w:3` is to insert padding when necessary. #1820 requested to add an option to define custom character, instead of space, e.g. `.eng(w:3, padding:'_')`, or something.

**Comment by virajbhartiya** [comment]
Hey @MaxVerevkin, I've made the changes to include custom padding.
Please have a look

**Comment by virajbhartiya** [comment]
Hey @MaxVerevkin, I've added custom padding.
Pease have a look

**Comment by MaxVerevkin** [comment]
Please test locally to avoid notification spam for those subscribed.

**Comment by virajbhartiya** [comment]
Hey @MaxVerevkin, Sorry for the extra commits.  It's passing the pre-commit checks.

Please have a look

**Comment by MaxVerevkin** [comment]
@virajbhartiya can you please squash and rebase to master?

**Comment by virajbhartiya** [comment]
@MaxVerevkin all the changes are in a single commit, Please review it

File: src/formatting/formatter.rs [hunk 1]
```diff
@@ -365,7 +378,7 @@ impl Formatter for EngFormatter {
 
                 let mut retval = match self.0.width as isize - digits {
                     isize::MIN..=0 => format!("{}", val.floor()),
-                    1 => format!(" {}", val.floor() as i64),
+                    1 => format!("{}", val.floor() as i64),
```
Comment (MaxVerevkin) [inline src/formatting/formatter.rs:381]: See my previous comment.

Comment (virajbhartiya) [reply src/formatting/formatter.rs]: added the default `' '`

Comment (MaxVerevkin) [reply src/formatting/formatter.rs]: I thought the point of this PR was to make this`' '` configurable... also I would prefer if this PR only affected the `.eng` formatter, not `.str`, unless you have a usecase for strings.

**Pull request closed without merging**